cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -907,21 +994,190 @@ class CastNode extends Node {
+  TSummaryCall(
+    FlowSummaryImpl::Public::SummarizedCallable c, FlowSummaryImpl::Private::SummaryNode receiver
+  ) {
+    FlowSummaryImpl::Private::summaryCallbackRange(c, receiver)
+  }
+
+/**
+ * A function call relevant for data flow. This includes calls from source
+ * code and calls inside library callables with a flow summary.
+ */
+class DataFlowCall extends TDataFlowCall {
+  /**
+   * Gets the underlying data flow call instruction, if any.
+   */
+  CallInstruction asCallInstruction() { none() }
+
+  /**
+   * Gets the operand the specifies the target function of the call.
+   */
+  CallTargetOperand getCallTargetOperand() { none() }
+
+  /**
+   * Gets the `Function` that the call targets, if this is statically known.
+   */
+  DataFlowCallable getStaticCallTarget() { none() }
+
+  /**
+   * Gets the `index`'th argument operand. The qualifier is considered to have index `-1`.
+   */
+  ArgumentOperand getArgumentOperand(int index) { none() }
+
+  /**
+   * Gets the argument at the specified index, or `this` if `index` is `-1`.
+   */
+  pragma[noinline]
+  final Instruction getArgument(int index) { result = this.getArgumentOperand(index).getDef() }
+
+  /**
+   * Gets the number of arguments of the call, including the `this` pointer, if any.
+   */
+  final int getNumberOfArguments() { result = count(this.getArgumentOperand(_)) }
+
+  /**
+   * Gets the enclosing callable, if any.
+   */
+  DataFlowCallable getEnclosingCallable() { none() }
+
+  /**
+   * Gets a textual representation of this call.
+   */
+  string toString() { none() }
+
+  /**
+   * Gets the location of this call.
+   */
+  Location getLocation() { none() }
+}
+
+/**
+ * TODO: QLDoc.
+ */
+private class NormalCall extends DataFlowCall, TNormalCall {
+  private CallInstruction call;
+
+  NormalCall() { this = TNormalCall(call) }
+
+  override CallInstruction asCallInstruction() { result = call }
+
+  override CallTargetOperand getCallTargetOperand() { result = call.getCallTargetOperand() }
+
+  override DataFlowCallable getStaticCallTarget() {
+    result = TSourceCallable(call.getStaticCallTarget())
+  }
+
+  override ArgumentOperand getArgumentOperand(int index) { result = call.getArgumentOperand(index) }
+
+  override DataFlowCallable getEnclosingCallable() {
+    result = TSourceCallable(call.getEnclosingFunction())
+  }
+
+  override string toString() { result = call.toString() }
+
+  override Location getLocation() { result = call.getLocation() }
+}
+
+/**
+ * A synthesized call inside a callable with a flow summary.
+ *
+ * TODO: example.
+ */
+class SummaryCall extends DataFlowCall, TSummaryCall {
+  private FlowSummaryImpl::Public::SummarizedCallable c;
+  private FlowSummaryImpl::Private::SummaryNode receiver;
+
+  SummaryCall() { this = TSummaryCall(c, receiver) }
+
+  /**
+   * Gets the data flow node that this call targets.
+   * OR
+   * Gets the data flow node that receives the result of the call.
+   * TODO: which is right?
```

Comment:
`/** Gets the data flow node that this call targets. */` would be the most accurate description.

Reply:
`receiver` is the name we use for a dataflow node that holds the "address" of a function (see my answer [here](https://github.com/github/codeql/pull/15371#discussion_r1519739461)). In particular, this is different from a "call target" because a call target (as you say) is a `Function`, and a `receiver` is a dataflow node that holds the "address" of the function.

It's actually used in other dataflow-related places as well - in particular for computing call targets for lambdas: https://github.com/github/codeql/blob/dddba3228b38f353f431976c42b6d0f83b25f3c2/shared/dataflow/codeql/dataflow/DataFlow.qll#L293

I agree that the terminology should be better explained. Possibly [here](https://github.com/github/codeql/blob/main/docs/ql-libraries/dataflow/dataflow.md#first-class-functions)